scripts/withRedux.js

```diff
@@ -0,0 +1,110 @@
+// => Reducer function <=
+function listReducer(tasks = [], action) {
+    switch (action.type) {
+        case "addTask":
+            let id = tasks.reduce((maxId, {id}) => id > maxId ? id : maxId, 0) + 1;
+            tasks.push({
+                id: tasks.length ? id : 0,
+                value: action.task
+            });
+            return tasks;
+        case "deleteTask":
+            return tasks.filter(task => task.id !== action.id);
+        default:
+            return tasks;
+    }
+}
+
+
+// => Rendering function <=
+
+// Version 1
+function listRender1() {
+    const tasks = store.getState();
+    const html = tasks.map(task => 
+        `<li id="${task.id}" class="to-do-item">
+            <input type="checkbox">
+            <label>${task.value}</label>
+            <button class="delete-button">
+                <img src="images/remove.png">
+            </button>
+        </li>`).join("");
+    list.innerHTML = html;
+    list.querySelectorAll("button").forEach(button => {
+        button.addEventListener("click", () => {
+            store.dispatch({id: parseInt(button.parentElement.getAttribute("id"), 10), type: "deleteTask"});
+        });
+    });
+}
+
+// Version 2
+function listRender2() {
```

Comment:
Very good! This is a more complex, but more efficient solution than version 1 - more logical checks, but fewer DOM updates, which are the more expensive operation. Very insightful - you will learn later on that e.g. React's reconciliation algorithm does pretty much the same thing, but on a higher level of complexity.